services/background.py

```
@@ -57,6 +58,20 @@ async def stop(self) -> None:
+        # Stop the threaded autonomy engines (best-effort).
+        for getter in ("get_self_healing_agent", "get_improvement_loop"):
+            try:
+                if getter == "get_self_healing_agent":
+                    from agent.self_healing import get_self_healing_agent as _g
+                else:
+                    from agent.improvement_loop import get_improvement_loop as _g
+                inst = _g()
+                if inst is not None:
+                    inst.stop()
+            except Exception:  # noqa: BLE001 — shutdown is best-effort
+                pass
```

Ruff (0.15.17)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)